sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java

```diff
@@ -3,83 +3,402 @@
-        return azureClient;
+    @NotNull
+    @Override
+    public Iterator<ClientRegistration> iterator() {
+        return Collections.singleton(azureClientRegistration.getClient()).iterator();
     }
 
-    public boolean isClientNeedConsentWhenLogin(ClientRegistration client) {
-        return otherClients.contains(client)
-            && properties.getAuthorizationClients().get(client.getClientName()) != null
-            && !properties.getAuthorizationClients().get(client.getClientName()).isOnDemand();
+    public AzureClientRegistration getAzureClientRegistration() {
+        return azureClientRegistration;
     }
 
-    public boolean isClientNeedConsentWhenLogin(String id) {
-        ClientRegistration client = findByRegistrationId(id);
-        return client != null && isClientNeedConsentWhenLogin(client);
+    public boolean isAzureDelegatedClientRegistrations(ClientRegistration client) {
+        return delegatedClientRegistrations.containsValue(client);
     }
 
-    public static boolean isDefaultClient(ClientRegistration clientRegistration) {
-        return AZURE_CLIENT_REGISTRATION_ID.equals(
-            clientRegistration.getClientName());
+    public boolean isAzureDelegatedClientRegistrations(String registrationId) {
+        return delegatedClientRegistrations.containsKey(registrationId);
     }
 
     public boolean isClientCredentials(String registrationId) {
-        return Optional.ofNullable(allClients.get(registrationId))
+        return Optional.ofNullable(allClientRegistrations.get(registrationId))
              .map(ClientRegistration::getAuthorizationGrantType)
              .map(AuthorizationGrantType::getValue)
              .map(v -> v.equals(AADAuthorizationGrantType.CLIENT_CREDENTIALS.getValue()))
              .orElse(false);
     }
 
-    public static boolean isDefaultClient(String clientId) {
-        return AZURE_CLIENT_REGISTRATION_ID.equals(clientId);
+    public static boolean isDefaultClient(String registrationId) {
+        return AZURE_CLIENT_REGISTRATION_ID.equals(registrationId);
+    }
+
+
+    public boolean isWebAppClientMode() {
+        boolean isWebAppClient = false;
+        if (isOAuth2ClientAvailable()
+            && !isResourceServerAvailable()
+            && StringUtils.hasText(properties.getClientId())
+            && !properties.getEnableWebAppResourceServer()) {
+            isWebAppClient = true;
+            LOGGER.debug("Apply the Web Application scenario.");
+        }
+        return isWebAppClient;
+    }
+
+
+    private boolean isWebApiClientMode() {
+        boolean isWebApiClient = false;
+        if (isOAuth2ClientAvailable()
+            && isResourceServerAvailable()
+            && !properties.getEnableWebAppResourceServer()) {
+            LOGGER.debug("Apply the Web Api scenario.");
+            isWebApiClient = true;
+        }
+        return isWebApiClient;
+    }
+
+    public boolean isAllInClientMode() {
```

Comment:
isWebApplicationAndResourceServer

Reply:
is it necessary to change the property `enableWebAppResourceServer` -> `enableWebAppAndResourceServer` ?